File: frontend/src/types/student.ts
import { VaccinationRecord } from '.';

export interface Student {
    id: string; // MongoDB ObjectId
    studentId: string;
    firstName: string;
    lastName: string;
    grade: string;
    section: string;
    dateOfBirth: string;
    parentName: string;
    contactNumber: string;
    vaccinationRecords: VaccinationRecord[];
}

export interface StudentFormData {
    studentId: string;
    firstName: string;
    lastName: string;
    grade: string;
    section: string;
    dateOfBirth: string;
    parentName: string;
    contactNumber: string;
} 

File: backend/src/main/java/com/vaccination/tracker/model/Student.java
package com.vaccination.tracker.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDate;
import java.util.List;

@Data
@Document(collection = "students")
public class Student {
    @Id
    private String id;
    private String studentId;
    private String firstName;
    private String lastName;
    private String grade;
    private String section;
    private LocalDate dateOfBirth;
    private String parentName;
    private String contactNumber;
    private List<VaccinationRecord> vaccinationRecords;
    
    @Data
    public static class VaccinationRecord {
        private String vaccineName;
        private LocalDate vaccinationDate;
        private String driveId;
        private String administeredBy;
    }
} 